src/config/configLoader.ts

```diff
@@ -1,59 +1,28 @@
-import { IParserResult } from '@stoplight/types';
-import { existsSync } from 'fs';
-import { isEmpty, merge } from 'lodash';
-import * as path from 'path';
-import { readParsable } from '../fs/reader';
-import { ConfigCommand, ConfigFormat, IConfig } from '../types/config';
+import { join } from '@stoplight/path';
+import * as fs from 'fs';
+import { ConfigFormat, ILintConfig } from '../types/config';
 
-const DEFAULT_CONFIG_FILE = 'spectral.yml';
+const DEFAULT_RULESET_FILE = /^\.?spectral\.(?:ya?ml|json)$/;
```

Comment:
I think that we should allow for an optional extra identifier between `spectral` and the extension, e.g. `spectral.foo.json`. Studio / graphite will likely need this.

Reply:
> I think that we should allow for an optional extra identifier between spectral and the extension, e.g. spectral.foo.json. Studio / graphite will likely need this.

I know we discussed it and I kept that in mind during the implementation, but now that I think of it I'm leaning towards stricter version, without the extra identifier in the middle.
Having an identifier may lead to a situation where renaming (`0` -> `1` etc.) might result in Spectral picking up a different ruleset. 
Are you fine with that?
I'm not very opinionated in terms of naming, happy to add the support for extra identifier, just beforehand I would like to make sure everyone is aware of possible consequences.

Reply:
> Having an identifier may lead to a situation where renaming (0 -> 1 etc.) might result in Spectral picking up a different ruleset.

Could you explain a little more? To put some more context to it, depending on the studio implementation, folks might have multiple spectral config files (one for oas2, one for oas3, etc). Graphite would handle reading them and feeding them to spectral instances, etc. This is not thought out, just an example of where it could be useful.

Reply:
Can we make a follow up 4.0.1 for this? The general public will need this less than studio or graphite and they don’t need it immediately either.

Reply:
> Could you explain a little more? 

if you have a file named `spectral.1.yaml` and create a new file called `spectral.0.yaml`, it's latter will be recognized as a default config if it's listed first.

Reply:
Ah ok got it. I was mistaken, you're correct - only `spectral.{json|yml|yaml}` should be considered for the default file.